ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -824,6 +839,12 @@ public void alterTable(String catName, String dbName, String tblName, Table newT
       throw new HiveException("Unable to alter table. " + e.getMessage(), e);
     } catch (TException e) {
       throw new HiveException("Unable to alter table. " + e.getMessage(), e);
+    } finally {
+      if (txnOpened) {
+        if (SessionState.get().getTxnMgr().isTxnOpen()) {
+          SessionState.get().getTxnMgr().commitTxn();
```

Comment:
Why do not we commit the TXN at the same place where we do it for the normal (DML) queries?

Reply:
I did not get your comment. In DMLs as well, they check the value of "transactional" and then open a transaction within the same method.

Reply:
Here you commit the TXN, but we commit the txn at the end of the query as well at Driver.releaseLocksAndCommitOrRollback(). I do not like that we commit the transaction here.

Reply:
The TXN is committed in Driver.releaseLocksAndCommitOrRollback(). I think it can cause problem if we commit them here too.